### ISSUE TITLE
New Rule: Should warn if no space following "USING"

### ISSUE THREAD

**Issue by barrywhart** [alanmcruickshank/sqlfluff#202]
Input:
```
SELECT user_id
FROM `t1`
JOIN `t2`
    USING(user_id)
```

No warning is produced. I think there should be a warning since the keyword `USING` is being written as if it were a function call. This is somewhat similar to #8, where we added a warning if `DISTINCT` was used with parentheses. Other cases may also arise. Maybe `ON`, for example?

**Comment by alanmcruickshank** [comment]
Nice - I really like this idea. I think this is probably something that I'm sloppy about at the moment. 👍
